File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <bits/stdc++.h>
using namespace std;

int a[9], sum;

int main() {
	//7난쟁이 키의 합 = 100
	//9C7의 합 == 100

	for (int i = 0; i < 9; i++) {
		cin >> a[i];//9난쟁이들의 키
	}

	sort(a, a + 9);
	do {
		sum = 0;
		for (int i = 0; i < 7; i++) sum += a[i];
		if (sum == 100) {
			for (int j = 0; j < 7; j++) cout << a[j] << '\n';
			break;
		}
	} while (next_permutation(a, a + 9));

	return 0;
}

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n').map(Number)

let sum = 0;

// 전체 합 - 9C2 === 100
let indexX = 0;
let indexY = 0;
let flag = 0;

input.sort((a, b) => a-b)

for(let i=0; i<9; i++){
  sum += input[i];
}


for(let i=0; i<9; i++){
  for(let j=i+1; j<9; j++){
    if(sum - input[i] - input[j] === 100){
      indexX = i; indexY = j;
      flag=1;
      break;
    }
  }
  if(flag) break;
}

for(let i=0; i<9; i++){
  if(i === indexX || i === indexY) continue;
  console.log(input[i])
}